src/entities/Proposal/jobs.ts

```diff
@@ -236,3 +180,49 @@ export async function publishBids(context: JobContext) {
     ErrorService.report('Error publishing bids', { error, category: ErrorCategory.Job })
   }
 }
+
+async function updateProposalsAndBudgets(proposalsWithOutcome: ProposalWithOutcome[], budgetsWithUpdates: Budget[]) {
+  const pool = new Pool({ connectionString: process.env.CONNECTION_STRING })
+  const client = await pool.connect()
+
+  try {
+    await client.query('BEGIN')
+
+    if (proposalsWithOutcome.length === 0) return
```

Comment:
We should return before opening a new pool connection, at the start of the function.